### PR TITLE
EXAMPLES/UCP: fixed hang for RNDV case in UCP client server example

### PR DESCRIPTION
**Pull request by karasevb**

### CODE REVIEW

**Review by brminich — APPROVED** [review]

File: examples/ucp_client_server.c [hunk 1]
```diff
@@ -448,7 +448,7 @@ static int send_recv_am(ucp_worker_h ucp_worker, ucp_ep_h ep, int is_server,
     if (is_server) {
         am_data_desc.recv_buf = msg;
 
-        while (!am_data_desc.complete) {
+        while (!am_data_desc.complete && !am_data_desc.is_rndv) {
```
Comment (yosefe) [inline examples/ucp_client_server.c:451]: 1. pls add a comment
2. what if is_rndv=1 from the previous iteration? can we exit before descriptor ready?

Comment (karasevb) [reply examples/ucp_client_server.c]: Fixed if is_rndv=1 from prev iteration. 
`am_data_desc.complete` means that the AM callback was called, but rndv is in progress yet.

**Review by brminich — APPROVED** [review]

File: examples/ucp_client_server.c [hunk 2]
```diff
@@ -416,7 +419,7 @@ ucs_status_t ucp_am_data_cb(void *arg, const void *header, size_t header_length,
 
 out:
     am_data_desc.complete = 1;
-    return UCS_OK;
+    return ret;
```
Comment (yosefe) [inline examples/ucp_client_server.c:422]: - set am_data_desc.complete = 1; before line 403
- line 410 jusr return INPROGRESS
- line 420,396 return UCS_OK
- remove label "out"

Comment (karasevb) [reply examples/ucp_client_server.c]: Done

**Review by yosefe — APPROVED** [review]
pls squash

**Pull request merged**